add-interactive.c

```diff
@@ -186,3 +244,82 @@ static struct file_stat **list_modified(struct repository *r,
 	free(s);
 	return files;
 }
+
+static void populate_wi_changes(struct strbuf *buf, struct adddel *ad,
+					char *no_changes)
+{
+	if (ad->add || ad->del)
+		strbuf_addf(buf, "+%"PRIuMAX"/-%"PRIuMAX,
+			    ad->add, ad->del);
+	else
+		strbuf_addf(buf, "%s", _(no_changes));
+}
+
+static struct choices *list_and_choose(struct choices *data,
+				       struct list_and_choose_options *opts)
+{
+	if (!data)
+		return NULL;
+
+	while (1) {
+		int i;
+		int last_lf = 0;
+		struct strbuf print;
+		struct strbuf index_changes;
+		struct strbuf worktree_changes;
+
+		strbuf_init(&print, 0);
+		strbuf_init(&index_changes, 0);
+		strbuf_init(&worktree_changes, 0);
+
+		if (opts->header.len) {
+			const char *header_color = get_color(COLOR_HEADER);
+			if (opts->header_indent)

```

Comment:
@dscho Here I got rid of `opts->list_flat` and used only `opts->header_indent`, as suggested. The point is to allow different levels of indentation (not depending on `list_flat`). But I am a bit worried about semantics. In Perl script, `list_flat == 0` means that `HEADER_INDENT` should be printed. Here, we don't have knowledge about `list_flat`, we just set or don't set `header_indent`. Since I replaced `list_flat` with `column_n` in C version, maybe I should add a comment in `list_and_choose` options saying that if we have `columns_n != 0`, we don't use `header_indent`? 
The whole comment is here: [https://public-inbox.org/git/pull.103.v4.git.gitgitgadget@gmail.com/T/#m7f7c705d69ba8f16b4ea2b1caedb82eefed58c97](https://public-inbox.org/git/pull.103.v4.git.gitgitgadget@gmail.com/T/#m7f7c705d69ba8f16b4ea2b1caedb82eefed58c97)

Reply:
It might make more sense to change `header_indent` from a flag to the actual indentation to use, i.e. a `const char *`.

Reply:
(To clarify: this `header_indent` would then be passed to `fputs()`, not `HEADER_INDENT`, and that constant would be dropped.)

Reply:
Sorry for being unclear, what I meant is that `header_indent` is actually `const char *` here. I just wanted to see if it's okay to see `list_flat` not having relations with `header_indent` anymore.

Reply:
From my perspective, that is perfectly fine.